### PR TITLE
Add check for uniqueness of list parameters

### LINKED ISSUE #1349: Visualization postprocessor does not check for unique plugins
When selecting the same visualization plugin more than once in release mode there is no error message, and the paraview output files are corrupted. Deal.II has an internal Assert, but since this is an issue in the input file, we should add an AssertThrow to catch it. This is mostly a reminder for me to come back to the issue.

### PR DESCRIPTION
**Pull request by gassmoeller**
This should fix #1349. I made a test for one of the plugins to test the new function, I guess we do not need one for every plugin that does the check.

### CODE REVIEW

**Comment by tjhei** [comment]
I don't particularly like the name ``list_is_unique`` because the argument is a ``std::vector``. Can you think of a better name? Maybe ``has_unique_entries`` or something? Looks good otherwise.

File: source/heating_model/interface.cc [hunk 1]
```diff
@@ -171,6 +171,11 @@ namespace aspect
         model_names
           = Utilities::split_string_list(prm.get("List of model names"));
 
+        AssertThrow(Utilities::list_is_unique(model_names),
+                    ExcMessage("The list of strings for the parameter "
+                               "'Heating model/List of model names' is not unique. "
```
Comment (bangerth) [inline source/heating_model/interface.cc]: The list is unique alright -- it's just that some entries appear twice :-)

How about "The list of strings for the parameter ... contains entries more than once. This is not allowed."?

File: source/utilities.cc [hunk 2]
```diff
@@ -861,6 +861,13 @@ namespace aspect
         return "";
     }
 
+    bool
+    list_is_unique (const std::vector<std::string> &list_of_strings)
+    {
+      const std::set<std::string> set_of_strings(list_of_strings.begin(),list_of_strings.end());
+      return (set_of_strings.size() == list_of_strings.size());
```
Comment (bangerth) [inline source/utilities.cc]: Clever :-)

**Comment by gassmoeller** [comment]
I addressed the comments and fixed the two tests that actually selected a postprocessor twice. Lets see if the tester is happy.

**Comment by bangerth** [comment]
OK to merge once tester is happy!

**Pull request merged**